### DISCUSSION TITLE
Store and Forward

### DISCUSSION THREAD

**Discussion by m2b** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#43] [Q&A] [answered]
How does this work?   What is the meaning of SCADA name?

**Comment by thomassorensen2650** [comment, marked as answer]
The primary SCADA has been renamed to Primary Host application in Sparkplug B v3.0. It's described in section 3.5. in [the specification](https://www.eclipse.org/tahu/spec/sparkplug_spec.pdf).

Basically the destination node that node-red will send the data to (destination can be any historian or SCADA system that supports this functionality) will publish a retained ONLINE message to the broker, with a DEATH certificate to set it OFFLINE. That way if the destination is unavailable (crashed or restarted or whatever), the node can buffer the data until the destination is back online. It can also be used for redundancy (this feature is not currently supported by this node-red node.

The documentation definitely need to be updated for this feature.
